File: VB/NewObjectInit.Module/NewValuesController.vb
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.SystemModule
Imports DevExpress.Persistent.BaseImpl

Namespace NewObjectInit.Module
    Partial Public Class NewValuesController
        Inherits ViewController

        Public Sub New()
            InitializeComponent()
            RegisterActions(components)
        End Sub
        Private detailViewController As ModificationsController
        Private Sub ViewController1_Activated(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Activated
            detailViewController = Frame.GetController(Of ModificationsController)()
            AddHandler detailViewController.SaveAndNewAction.Executing, AddressOf SaveAndNewAction_Executing
            AddHandler Frame.ViewChanged, AddressOf Frame_ViewChanged
        End Sub

        Private Sub Frame_ViewChanged(ByVal sender As Object, ByVal e As ViewChangedEventArgs)
            If person IsNot Nothing AndAlso View.CurrentObject IsNot Nothing AndAlso View.CurrentObject.GetType() Is GetType(Person) Then
                Dim current As Person = CType(View.CurrentObject, Person)
                current.FirstName = person.FirstName
                current.LastName = person.LastName
                person = Nothing
            End If
        End Sub

        Private Shared person As Person

        Private Sub SaveAndNewAction_Executing(ByVal sender As Object, ByVal e As CancelEventArgs)
            If View.CurrentObject.GetType() Is GetType(Person) Then
                person = CType(View.CurrentObject, Person)
            End If
        End Sub


    End Class
End Namespace

File: CS/NewObjectInit.Module/NewValuesController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.SystemModule;
using DevExpress.Persistent.BaseImpl;

namespace NewObjectInit.Module
{
    public partial class NewValuesController : ViewController
    {
        public NewValuesController()
        {
            InitializeComponent();
            RegisterActions(components);
        }

        DetailViewController detailViewController;

        private void ViewController1_Activated(object sender, EventArgs e)
        {
            detailViewController = Frame.GetController<DetailViewController>();
            detailViewController.SaveAndNewAction.Executing += new CancelEventHandler(SaveAndNewAction_Executing);
            Frame.ViewChanged += new EventHandler<ViewChangedEventArgs>(Frame_ViewChanged);
        }

        void Frame_ViewChanged(object sender, ViewChangedEventArgs e)
        {
            if (person != null && View.CurrentObject != null && View.CurrentObject.GetType() == typeof(Person))
            {
                Person current = (Person)View.CurrentObject;
                current.FirstName = person.FirstName;
                current.LastName = person.LastName;
                person = null;
            }
        }

        static Person person;

        void SaveAndNewAction_Executing(object sender, CancelEventArgs e)
        {
            if (View.CurrentObject.GetType() == typeof(Person))
            {
                person = (Person)View.CurrentObject;
            }
        }


    }
}
